cnd/src/db/custom_sql_types.rs

```diff
@@ -0,0 +1,78 @@
+use diesel::{
+    backend::Backend,
+    deserialize::{self, FromSql},
+    serialize::{self, Output, ToSql},
+    sql_types,
+};
+use std::{convert::TryFrom, fmt, ops::Deref, str::FromStr};
+
+/// Custom diesel new-type that works as long as T implements `Display` and
+/// `FromStr`.
+#[derive(Debug, Clone, Copy, PartialEq, FromSqlRow, AsExpression)]
+#[sql_type = "sql_types::Text"]
+pub struct Text<T>(pub T);
+
+impl<T> Deref for Text<T> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<DB, T> ToSql<sql_types::Text, DB> for Text<T>
+where
+    DB: Backend,
+    String: ToSql<sql_types::Text, DB>,
+    T: fmt::Display + fmt::Debug,
+{
+    fn to_sql<W: std::io::Write>(&self, out: &mut Output<'_, W, DB>) -> serialize::Result {
+        let s = self.0.to_string();
+        s.to_sql(out)
+    }
+}
+
+impl<DB, T> FromSql<sql_types::Text, DB> for Text<T>
+where
+    DB: Backend,
+    String: FromSql<sql_types::Text, DB>,
+    T: FromStr,
+    <T as FromStr>::Err: std::error::Error + Send + Sync + 'static,
+{
+    fn from_sql(bytes: Option<&DB::RawValue>) -> deserialize::Result<Self> {
+        let s = String::from_sql(bytes)?;
+        let parsed = T::from_str(s.as_ref())?;
+
+        Ok(Text(parsed))
+    }
+}
+
+// Custom diesel new type for enforcing storage of a u32
```

Comment:
:upside_down_face: 
```suggestion
// Custom diesel newtype for enforcing storage of a u32
```

Reply:
Please open a follow-up PR for these kind of changes! I don't want to wait for another build to land it :see_no_evil: